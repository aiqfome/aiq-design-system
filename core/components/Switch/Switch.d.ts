import React, { InputHTMLAttributes } from 'react';
import { DefaultTheme } from 'styled-components';
export interface Props extends DefaultTheme, InputHTMLAttributes<HTMLInputElement> {
    checked?: boolean;
    disabled?: boolean;
    className?: string;
    variant?: 'default' | 'small';
}
export declare const Switch: React.ForwardRefExoticComponent<Props & React.RefAttributes<HTMLInputElement>>;
//# sourceMappingURL=Switch.d.ts.map