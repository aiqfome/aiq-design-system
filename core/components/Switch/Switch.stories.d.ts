import React, { ReactElement } from 'react';
declare const _default: {
    component: React.ForwardRefExoticComponent<import("./Switch").Props & React.RefAttributes<HTMLInputElement>>;
    title: string;
    decorators: ((...args: any) => any)[];
};
export default _default;
export declare const Basic: () => ReactElement;
export declare const BasicDisabled: () => ReactElement;
export declare const Checked: () => ReactElement;
export declare const DisabledChecked: () => ReactElement;
export declare const Variants: () => ReactElement;
//# sourceMappingURL=Switch.stories.d.ts.map