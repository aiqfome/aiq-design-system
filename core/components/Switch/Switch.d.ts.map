{"version":3,"file":"Switch.d.ts","sourceRoot":"","sources":["../../../src/components/Switch/Switch.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,mBAAmB,EAAE,MAAM,OAAO,CAAA;AAElD,OAAe,EAAO,YAAY,EAAE,MAAM,mBAAmB,CAAA;AAG7D,MAAM,WAAW,KACf,SAAQ,YAAY,EAClB,mBAAmB,CAAC,gBAAgB,CAAC;IACvC,OAAO,CAAC,EAAE,OAAO,CAAA;IACjB,QAAQ,CAAC,EAAE,OAAO,CAAA;IAClB,SAAS,CAAC,EAAE,MAAM,CAAA;IAClB,OAAO,CAAC,EAAE,SAAS,GAAG,OAAO,CAAA;CAC9B;AA6GD,eAAO,MAAM,MAAM,gFAYlB,CAAA"}