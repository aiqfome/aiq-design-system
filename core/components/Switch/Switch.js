"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Switch = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
const switchVariations = {
    default: styled_components_1.css `
    & > input {
      width: 40px;
      height: 15px;

      &::after {
        margin-top: -3px;
        width: 21px;
        height: 21px;
        right: 19px;
      }
    }
  `,
    small: styled_components_1.css `
    & > input {
      width: 25px;
      height: 10px;

      &::after {
        margin-top: -2px;
        width: 14px;
        height: 14px;
        right: 11px;
      }
    }
  `
};
const SwitchStyled = styled_components_1.default.label `
  ${styled_system_1.space}

  display: flex;
  align-items: center;

  ${({ variant }) => switchVariations[variant || 'default']}

  & > input {
    cursor: pointer;
    position: relative;
    margin: 0 10px 0 0;

    &::before,
    &::after {
      content: '';
      position: absolute;
    }

    &::before {
      width: 100%;
      background-color: ${({ theme }) => theme.colors.grey};
      border-radius: 30px;
      height: 100%;
      transition: all 0.25s ease-in-out;
    }

    &::after {
      background-color: ${({ theme }) => theme.colors.lightGrey};
      border-radius: 21px;
      box-shadow: 0px 0px 1px 0px rgba(0, 0, 0, 0.75);
      transition: all 0.2s;
    }

    &:not(:checked):not(:disabled):hover::before {
      background-color: ${({ theme }) => theme.colors.primaryLight};
    }

    &:checked::after {
      right: 0;
    }

    &:checked::before {
      background-color: ${({ theme }) => theme.colors.primary};
    }

    &:disabled {
      cursor: default;
    }

    &:disabled::after {
      box-shadow: none;
      background-color: ${({ theme }) => theme.colors.mediumGrey};
    }

    &:disabled::before {
      background-color: ${({ theme }) => theme.colors.mediumGrey};
      border-color: rgba(0, 0, 0, 0.12);
      box-shadow: none;
    }

    &:disabled:checked::before {
      background-color: ${({ theme }) => theme.colors.mediumGrey};
    }

    &:disabled:checked:after {
      background-color: ${({ theme }) => theme.colors.primaryLight};
      box-shadow: none;
    }

    label {
      display: block;
      margin-bottom: 10px;
      cursor: pointer;
    }
  }
`;
exports.Switch = react_1.default.forwardRef((_a, ref) => {
    var { variant = 'default', className } = _a, props = __rest(_a, ["variant", "className"]);
    return (react_1.default.createElement(SwitchStyled, { variant: variant, className: className, "data-testid": 'switch' },
        react_1.default.createElement("input", Object.assign({ ref: ref, type: 'checkbox', "data-testid": 'input' }, props))));
});
exports.Switch.displayName = 'Switch';
exports.Switch.propTypes = {
    checked: prop_types_1.default.bool,
    variant: prop_types_1.default.oneOf(['default', 'small']),
    className: prop_types_1.default.string
};
