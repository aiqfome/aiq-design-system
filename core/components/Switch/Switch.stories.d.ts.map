{"version":3,"file":"Switch.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Switch/Switch.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,YAAY,EAAY,MAAM,OAAO,CAAA;;;;;;AAOrD,wBAIC;AAED,eAAO,MAAM,KAAK,QAAO,YAaxB,CAAA;AAED,eAAO,MAAM,aAAa,QAAO,YAShC,CAAA;AAED,eAAO,MAAM,OAAO,QAAO,YAa1B,CAAA;AAED,eAAO,MAAM,eAAe,QAAO,YAUlC,CAAA;AAED,eAAO,MAAM,QAAQ,QAAO,YAsB3B,CAAA"}