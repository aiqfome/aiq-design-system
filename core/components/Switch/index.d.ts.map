{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Switch/index.tsx"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAA"}