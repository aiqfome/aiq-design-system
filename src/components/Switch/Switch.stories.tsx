import React, { ReactElement } from 'react'

import { withKnobs, boolean } from '@storybook/addon-knobs'

import { Flex } from '../Flex'
import { Switch } from './Switch'

export default {
  component: Switch,
  title: 'Switch',
  decorators: [withKnobs]
}

export const Basic = (): ReactElement => {
  return (
    <Flex variant='fullCentralized'>
      <Switch checked={boolean('checked', false)} />
    </Flex>
  )
}

export const BasicDisabled = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Switch disabled />
  </Flex>
)

export const Checked = (): ReactElement => {
  return (
    <Flex variant='fullCentralized'>
      <Switch checked={true} />
    </Flex>
  )
}

export const DisabledChecked = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Switch disabled checked />
  </Flex>
)
