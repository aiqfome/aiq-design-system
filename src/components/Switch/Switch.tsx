import React, { useState } from 'react'
import PropTypes from 'prop-types'
import styled, { DefaultTheme } from 'styled-components'
import { space } from 'styled-system'

export interface Props extends DefaultTheme {
  checked?: boolean
  disabled?: boolean
}

const SwitchStyled = styled.label`
  ${space}

  & > input {
    cursor: pointer;
    width: 40px;
    height: 15px;
    position: relative;
    margin: 0 10px 0 0;

    &::before,
    &::after {
      content: '';
      position: absolute;
    }

    &::before {
      width: 100%;
      background-color: ${({ theme }) => theme.colors.grey};
      border-radius: 30px;
      height: 100%;
      transition: all 0.25s ease-in-out;
    }

    &::after {
      width: 21px;
      height: 21px;
      margin-top: -3px;
      background-color: ${({ theme }) => theme.colors.lightGrey};
      border-radius: 21px;
      right: 19px;
      box-shadow: 0px 0px 1px 0px rgba(0, 0, 0, 0.75);
      transition: all 0.2s;
    }

    &:not(:checked):not(:disabled):hover::before {
      background-color: ${({ theme }) => theme.colors.primaryLight};
    }

    &:checked::after {
      right: 0;
    }

    &:checked::before {
      background-color: ${({ theme }) => theme.colors.primary};
    }

    &:disabled {
      cursor: default;
    }

    &:disabled::after {
      box-shadow: none;
      background-color: ${({ theme }) => theme.colors.mediumGrey};
    }

    &:disabled::before {
      background-color: ${({ theme }) => theme.colors.white};
      border-color: rgba(0, 0, 0, 0.12);
      box-shadow: none;
    }

    &:disabled:checked::before {
      background-color: ${({ theme }) => theme.colors.mediumGrey};
    }

    &:disabled:checked:after {
      background-color: ${({ theme }) => theme.colors.primaryLight};
      box-shadow: none;
    }

    label {
      display: block;
      margin-bottom: 10px;
      cursor: pointer;
    }
  }
`

export const Switch: React.FC<Props> = ({
  checked = false,
  disabled = false
}: Props) => {
  const [isChecked, setIsChecked] = useState(checked)

  return (
    <SwitchStyled>
      <input
        type='checkbox'
        checked={isChecked}
        disabled={disabled}
        onChange={() => setIsChecked(!isChecked)}
      />
    </SwitchStyled>
  )
}

Switch.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool
}
