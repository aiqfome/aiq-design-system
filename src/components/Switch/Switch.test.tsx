import React from 'react'

import { Switch } from './Switch'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(<Switch />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
