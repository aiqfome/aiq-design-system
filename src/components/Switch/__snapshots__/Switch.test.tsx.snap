// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<label
  className="sc-AxjAm bUMAfg"
>
  <input
    type="checkbox"
  />
</label>
`;
